### ISSUE TITLE
The Filter module has a hidden dependency on the Link (field) module

### ISSUE THREAD

**Issue by indigoxela** [backdrop/backdrop-issues#4647]
**Description of the bug**

While testing for another PR I figured out that entering internal links into CKEditor depends on the Link module. Otherwise the links are all 404 when you're not on the front page.

**Steps To Reproduce**

1. Disable the Link module
2. Open the node form of any post ("Your first post" in my example)
3. Open the Link dialog in CKEditor and start typing to get an internal path
4. Insert and save the node
5. Follow the new link

**Actual behavior**

```
Page not found
The requested page "/posts/posts/your-first-post" could not be found.
```
When the Link module isn't enabled, the base path won't get added. And the relative path `posts/your-first-post` **when you're not on the front page** is wrong.

**Expected behavior**

The base_path gets added independently from the Link field module installed or not.

**Additional information**

This is [the code](https://github.com/backdrop/backdrop/blob/f4178d3babf538caba254281a6e82a460c3163e9/core/modules/filter/filter.pages.inc#L498)
See Issue #3083 for details.

**Proposed solution**

Find a replacement in Filter module for function link_validate_url() - which belongs to the Link module.

**Comment by indigoxela** [comment]
The problem with missing base path should be fixed in #4630.

**Comment by indigoxela** [comment]
This got fixed in #4630
